src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -1434,6 +1451,7 @@ void IDictionary.Add(object key, object value)
         {
             if (key == null) ThrowKeyNullException();
             if (!(key is TKey)) throw new ArgumentException(SR.ConcurrentDictionary_TypeOfKeyIncorrect);
+            ThrowIfNullAndNullsAreIllegal(value);
```

Comment:
I don't understand why this call is needed.  What's an example it impacts?

Reply:
Ah, it's here to ensure that a null for a value type TValue ends up with an ArgumentException instead of a NullReferenceException I guess.

Reply:
Yes. That was the reason.